app/views/users/show.html.erb

```diff
@@ -36,6 +36,18 @@
             <%= c.user_count%>
           </div>
         </div>
+        <div class='col'>
+          <div class='course-details_title'>
+            Role
+          </div>
+          <div class='course-details_value'>
+            <% r = c.courses_users %>
+            <% r.each do |p| %>
+              <% @pass = p.role %>
+            <%end%>
+            <%= role(@pass) %>
```

Comment:
This loops through each courses_user in the course `c`, reassigning `@pass` for each one, and then rendering the last one. That means courses_user may not be the same user that we want.

There are a few alternatives. Maybe the most straightforward one would be to set @courses_users instead of @course in the controller, and use that as the basis for the entire view. (You can get the course for a CoursesUsers record with courses_user.course.)